geode-core/src/distributedTest/java/org/apache/geode/cache/ConnectionPoolDUnitTest.java

```diff
@@ -1966,230 +1394,179 @@ public void run2() throws CacheException {
-        factory.setConcurrencyChecksEnabled(false);
-        ClientServerTestCase.configureConnectionPool(factory, host0, port, -1, true, -1, -1, null);
-        CertifiableTestCacheListener l = new CertifiableTestCacheListener(
-            org.apache.geode.test.dunit.LogWriterUtils.getLogWriter());
-        factory.setCacheListener(l);
-        Region rgn = createRegion(name, factory.create());
-        rgn.registerInterestRegex(".*", false, false);
-      }
-    };
-
-    vm1.invoke(createEmpty);
-    vm1.invoke("Populate region", new CacheSerializableRunnable() {
-      @Override
-      public void run2() throws CacheException {
-        Region<Object, Object> region = getRootRegion().getSubregion(name);
+    vm1.invoke("Create region", () -> {
+      getLonerSystem();
+      getCache();
+      RegionFactory<Object, Object> factory = getCache().createRegionFactory();
+      factory.setScope(Scope.LOCAL);
+      factory.setConcurrencyChecksEnabled(false);
+      configureConnectionPool(factory, host0, new int[] {port}, true, -1, -1, null);
+      factory.addCacheListener(new CertifiableTestCacheListener<>());
+      factory.setDataPolicy(DataPolicy.EMPTY);
+      factory.setSubscriptionAttributes(new SubscriptionAttributes(InterestPolicy.ALL));
+      Region rgn = createRegion(name, factory);
+      rgn.registerInterestRegex(".*", false, false);
+    });
+    vm1.invoke("Populate region", () -> {
+      Region<Object, Object> region = getRootRegion().getSubregion(name);
+      for (int i = 0; i < 10; i++) {
+        region.put(i, "old" + i);
+      }
+    });
+
+    vm2.invoke("Create region", () -> {
+      getLonerSystem();
+      getCache();
+      RegionFactory<Object, Object> factory = getCache().createRegionFactory();
+      factory.setScope(Scope.LOCAL);
+      factory.setConcurrencyChecksEnabled(false);
+      configureConnectionPool(factory, host0, new int[] {port}, true, -1, -1, null);
+      factory.addCacheListener(new CertifiableTestCacheListener<>());
+      Region rgn = createRegion(name, factory);
+      rgn.registerInterestRegex(".*", false, false);
+    });
+    vm1.invoke("Turn on history", () -> {
+      Region<Object, Object> region = getRootRegion().getSubregion(name);
+      CertifiableTestCacheListener<Object, Object> ctl =
+          (CertifiableTestCacheListener<Object, Object>) region.getAttributes()
+              .getCacheListeners()[0];
+      ctl.enableEventHistory();
+    });
+    vm2.invoke("Update region", () -> {
+      Region<Object, Object> region = getRootRegion().getSubregion(name);
+      for (int i = 0; i < 10; i++) {
+        region.put(i, "new" + i, "callbackArg" + i);
+      }
+    });
+
+    vm1.invoke("Verify invalidates", () -> {
+      Region<Object, Object> region = getRootRegion().getSubregion(name);
+      await().untilAsserted(() -> {
         for (int i = 0; i < 10; i++) {
-          region.put(i, "old" + i);
+          assertThat(region.get("new" + i) == ("callbackArg" + i));
```

Comment:
This compiles but does nothing. It's the same as saying:
```
assertThat(true);
or
assertThat(false);
```
....both of which will never fail. "assertThat(true).isTrue()" would work.

You should change this to:
```
assertThat(region.get("new" + i)).isEqualTo("callbackArg" + i);
```